### PR TITLE
Gets trace IDs from MDC

### PR DESCRIPTION
**Pull request by joaoandremartins**
Currently, we're extracting the trace IDs from the request headers.
This has several shortcomings, such as the trace IDs not being in sync
when Brave generates new ones.
The MDC should be the single source of truth for getting trace IDs.
It is already being populated by Spring Cloud Sleuth and we should
reuse that mechanism.

Because header extraction is already being performed by Spring Cloud
Sleuth, all this logic is eliminated from this project.
This leaves us unable to get trace IDs from requests within certain
GCP infrastructure that uses the "x-cloud-trace-context" header.
Because adding a new extractor to the zipkin propagation isn't trivial,
this work will need to be done on a separate PR.

Some documentation repairs also done, including eliminating repeated
logging module's README.

### CODE REVIEW

**Comment by artembilan** [comment]
Is Travis failure related to your changes somehow?

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverJsonLayout.java [hunk 1]
```diff
@@ -194,11 +154,14 @@ public void start() {
 			});
 		}
 		if (this.includeTimestamp) {
-			map.put(TIMESTAMP_SECONDS_ATTRIBUTE, TimeUnit.MILLISECONDS.toSeconds(event.getTimeStamp()));
-			map.put(TIMESTAMP_NANOS_ATTRIBUTE, TimeUnit.MILLISECONDS.toNanos(event.getTimeStamp() % 1_000));
+			map.put(StackdriverTraceConstants.TIMESTAMP_SECONDS_ATTRIBUTE,
+					TimeUnit.MILLISECONDS.toSeconds(event.getTimeStamp()));
+			map.put(StackdriverTraceConstants.TIMESTAMP_NANOS_ATTRIBUTE,
+					TimeUnit.MILLISECONDS.toNanos(event.getTimeStamp() % 1_000));
```
Comment (artembilan) [inline spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverJsonLayout.java]: Is it really correct to use here *Modulus Operator*, but not *Divide*?

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverTraceConstants.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.logging;
+
+/**
+ * @author João André Martins
+ */
+public class StackdriverTraceConstants {
```
Comment (artembilan) [inline spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverTraceConstants.java]: Make it `final`, please and with the `private` ctor: https://github.com/spring-projects/spring-framework/wiki/Code-Style#utility-classes

Not sure why the story there is about `abstract`, but I'm fine with `final`

**Comment by meltsufin** [comment]
@joaoandremartins Before I get too deep into this, is Logging correlation by trace ids still working even if  Cloud Trace / Sleuth are not used?

**Comment by joaoandremartins** [comment]
Logging correlation does not work well as it is.
That being said, it won't work without Sleuth, because Sleuth or Brave is the one extracting and writing the trace ID to the MDC. For that matter, neither does the JSON one currently, since it's getting the trace ID from the MDC as well.

I'm also not sure if non-Sleuth environments are actually something we should be worrying about on this **Spring** Cloud GCP repo.

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/TraceIdLoggingEnhancer.java [hunk 3]
```diff
@@ -19,37 +19,28 @@
...
-	 * @return the trace ID in the "projects/[MY_PROJECT_ID]/traces/[MY_TRACE_ID]"
-	 */
-	public static String getCurrentTraceId() {
-		return traceId.get();
-	}
+	private GcpProjectIdProvider projectIdProvider = new DefaultGcpProjectIdProvider();
 
 	@Override
 	public void enhanceLogEntry(LogEntry.Builder builder) {
-		String traceId = getCurrentTraceId();
```
[the remark this answers is not in the document]
Comment (joaoandremartins) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/TraceIdLoggingEnhancer.java]: That would require keeping all that extractor machinery around just to feed that alternative use case, which we saw is better served by just delegating to a Brave extractor.

[the remark this answers is not in the document]
Comment (meltsufin) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/TraceIdLoggingEnhancer.java]: Why not keep all the machinery around though? We already have it and it supports the use-case where one uses our logging support (log  grouping by request) without Sleuth/Trace. What's the issue with that? I don't think it takes anything away from the better support for Brave that you're implementing here.

**Comment by joaoandremartins** [comment]
Not removing the current logging module for now.
#714 was merged, so this can be closed.

**Pull request closed without merging**